src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/DashboardItemOrder.tsx

```diff
@@ -183,7 +183,7 @@ export const DashboardItemOrder = ({
                 </RouterButton>
               )}
             </div>
-            {!writable && !order.contract?.signed_on && (
+            {!writable && order.contract && !order.contract.signed_on && (
```

Comment:
What do you think to store this condition statement within a variable with an explicit name like `hasContractToSign` ?